src/libs/actions/Report.ts

```diff
@@ -2149,7 +2156,15 @@ function leaveRoom(reportID: string, isWorkspaceMemberLeavingWorkspaceRoom = fal
 
     API.write('LeaveRoom', parameters, {optimisticData, successData, failureData});
 
-    if (isWorkspaceMemberLeavingWorkspaceRoom) {
+    const lastAccessedReportID =
+        ReportUtils.sortReportsByLastRead(allReports, reportMetadata)
+            .filter((r) => r?.reportID !== reportID && r?.notificationPreference !== CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN)
+            .at(-1)?.reportID ?? null;
+    if (lastAccessedReportID) {
+        // We should call Navigation.goBack to pop the current route first before navigating to Concierge.
+        Navigation.goBack(ROUTES.HOME);
```

Comment:
Q: why do we need this?

Reply:
I just applied the same fix from https://github.com/Expensify/App/pull/33744 to avoid navigating to the room just left  when closing the current chat on mobile.